### ISSUE TITLE
Update hot-functions containers

### ISSUE THREAD

**Issue by sebgl** [iron-io/functions#612]
Hello,

Context:
I'm running a hot function with 30sec timeouts. It runs on events every 10sec. Hence the container never reaches the timeout and runs forever.

Problem:
I'm changing my function's code and pushing the updated docker image to a registry (same image name).
I can update the route using IronFunctions API (PATCH /apps/{app}/routes/{route}), but since the container never stops running, my new container will actually never be pulled again in its latest version. Somehow I would expect that IronFunctions would stop the container between two functions execution, and run the new one (by pulling the new image).

How can I manage my function update & redeployment without stopping the event flow? Maybe I missed something in the way hot functions can be used.

Thanks !

**Comment by pedronasser** [comment]
Good point, we should make some kind of routine that checks for image updates and without stopping the old version containers, the runner would start new containers with the new version and just redirect the requests for the new group of containers, then it can stop old containers.

@treeder @seiflotfy what do you think?
